snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshot.java

```diff
@@ -110,6 +131,26 @@ public long getChecksum() {
     return checksum;
   }
 
+  @Override
+  public ActorFuture<SnapshotReservation> reserve() {
+    final CompletableActorFuture<SnapshotReservation> snapshotLocked =
+        new CompletableActorFuture<>();
+    actor.run(
```

Comment:
:question: The usage of the actor scheduler looks a bit out of place here.  I guess it is necessary because multiple actors will access snapshots? Once through snapshot store, to delete older snapshots, and once through some backup actor to reserve snapshots?

Reply:
That's correct.

Reply:
Which other calls? Snapshot was always a shared resource, in that sense nothing has changed now.

Reply:
Sorry, what I mean is this: `FileBasedSnapshot#delete()` and others do not make any use of `actor.call` or similar methods, that happens one layer above, in the `FileBasedSnapshotStore`  via `FileBasedReceivedSnapshot.

Now, `FileBasedSnapshot#reserve` and `FileBasedSnapshot#removeReservation` make use of `actor.call`. IMO that looks confusing because it's not clear why some methods do it and some don't. That's why I was wondering if we could align everything so that `FileBasedSnapshot`'s public methods always submit to the actor scheduler.

Reply:
Yep, that sounds good to me :+1: